src/foam/u2/mlang/Pie.js

```diff
@@ -0,0 +1,38 @@
+foam.CLASS({
+  package: 'foam.u2.mlang',
+  name: 'Pie',
+  extends: 'foam.mlang.sink.GroupBy',
+  requires: [
+    'foam.graphics.DataSource',
+    'foam.graphics.PieGraph',
+  ],
+  properties: [
+    {
+      name: 'graphArgs',
+      value: {
+        radius: 200,
+        x: 150,
+        y: 150,
+        margin: 1.3,
+        graphColors: [ '#d81e05', '#093649', '#59a5d5', '#2cab70' ],
+      },
+    },
+    {
+      name: 'graph_',
+      expression: function(graphArgs, groups) {
+        var seriesValues = Object.values(groups).map(function(sink) {
+          return sink.value;
+        })
+        if ( ! seriesValues.length ) seriesValues = [0];
+        var p = this.PieGraph.create(graphArgs);
+        p.seriesValues = seriesValues;
+        return p;
+      },
+    },
+  ],
+  methods: [
+    function toE(_, x) {
+      return x.E().add(this.graph_$);
```

Comment:
Could you instead:
function toE(args, X) {return this.graph_.toE(args, X); }

Reply:
I need to use the graph_$ slot because the view can change based on the expression.

Reply:
ok